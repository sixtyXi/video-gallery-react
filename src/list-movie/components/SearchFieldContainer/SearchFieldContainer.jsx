import React, { Component } from 'react';

import styles from './SearchFieldContainer.scss';

export class SearchFieldContainer extends Component {
  state = {
    searchTxt: ''
  };

  handleChange = (event) => {
    this.setState({ searchTxt: event.target.value });
  }

  render() {
    const { searchTxt } = this.state;

    return (
      <div className={styles.searchField}>
        <input
          data-cy="searchField"
          type="text"
          value={searchTxt}
          id="search"
          onChange={this.handleChange}
        />
      </div>
    );
  }
}
