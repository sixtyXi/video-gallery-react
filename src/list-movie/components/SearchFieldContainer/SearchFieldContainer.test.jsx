import React from 'react';
import { shallow, mount } from 'enzyme';

import { SearchFieldContainer } from './SearchFieldContainer';

describe('SearchFieldContainer', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<SearchFieldContainer />);

    expect(wrapper).toMatchSnapshot();
  });

  test('calls *handleChange* on input value changes', () => {
    const wrapper = mount(<SearchFieldContainer />);
    const spy = jest.spyOn(wrapper.instance(), 'handleChange');
    const input = wrapper.find('input');

    input.simulate('change');
    expect(spy).toHaveBeenCalled();
  });

  test('sets new input value to state', () => {
    const wrapper = mount(<SearchFieldContainer />);
    const input = wrapper.find('input');

    input.simulate('change', { target: { value: 'New text' } });
    expect(wrapper.state().searchTxt).toBe('New text');
  });
});
