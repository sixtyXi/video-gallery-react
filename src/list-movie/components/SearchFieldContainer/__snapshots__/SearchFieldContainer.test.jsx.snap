// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFieldContainer renders correctly 1`] = `
<div
  className="searchField"
>
  <input
    id="search"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
